### ISSUE TITLE
Support the spread operator

### ISSUE THREAD

**Issue by hedyhli** [nushell/tree-sitter-nu#98]
<img width="771" alt="image" src="https://github.com/nushell/tree-sitter-nu/assets/50042066/26a08066-95de-47b5-8ce5-fd00aa4ee48f">

Currently, the `...` is parsed as a string, whereas it's a special operator that should be assigned something special, and then highlighted as an operator. If it's parsed as a `expr_unary`, `highlights.scm` can then be changed as follows:

```diff
diff --git a/queries/nu/highlights.scm b/queries/nu/highlights.scm
index c73f8f3..ce0b839 100644
--- a/queries/nu/highlights.scm
+++ b/queries/nu/highlights.scm
@@ -161,7 +161,7 @@
     "++="
 ] @operator)
 
-(expr_unary ["not" "-"] @operator)
+(expr_unary ["not" "-" "..."] @operator)
 
 (val_range [
     ".."
```

**Comment by fdncred** [comment]
agreed

**Comment by mrdgo** [comment]
So far, I couldn't get this to work in pure `grammar.js`. As I understand, the dots must have no whitespace before the list and we would have to detect a correctly delimited list. I fear that we need to implement this in a `scanner.c`. Except @blindFS can do some magic?

**Comment by blindFS** [comment]
I noticed this issue the other day, my plan is

1. create new rules for ...() ...[] ...{} ...$, correct me if something's missing. 
2. Then add those rules to allowed places, I believe ...{} is only allowed inside record.

We can group ...( for example as single terminal token, so it won't mess other patterns with leading dots like range or decimal.

**Comment by blindFS** [comment]
Maybe I should create a new issue for all the problems that I found so far, so that someone may come and fix it before me.

**Comment by blindFS** [comment]
> I'd take a look at this. I also observed a problem with an alias:
> 
> ![image](https://private-user-images.githubusercontent.com/1636639/370233502-82e3c90d-80fc-4707-b4c7-90ecd1e11c55.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE4MjMwMDksIm5iZiI6MTczMTgyMjcwOSwicGF0aCI6Ii8xNjM2NjM5LzM3MDIzMzUwMi04MmUzYzkwZC04MGZjLTQ3MDctYjRjNy05MGVjZDFlMTFjNTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTdUMDU1MTQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDEyMWQ2NjNkYjM4ODVhYzk1NWI1OWJhYjU4OTRjOGVlYzE0MjlhZDM5OGFmYmYzOTQ5ZjRmMWZlMjdhZWJkNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.VD0ogWAjNjjpDSVm8rqXElZUBEa6Og3PFumRIue6oHo)
> 
> They all work correctly, even without the quotes, but tree-sitter-nu can't handle them. These two are probably related. If not, I'll create separate PRs

@mrdgo The alias example still fails after #143, but it's another issue caused by restricted command identifier.
